### PR TITLE
375 docker config

### LINKED ISSUE #375: Enhancement: Add a Docker config to allow easy server launches
### Describe the Current Behavior/Feature:

Currently, the only means we provide for running PASS is to run the both the solid server and the react server directly, as separate processes. This is fine (and often preferable) for a dev environment, but becomes problematic when deploying the server in a live environment.

---
### Proposed Behavior/Feature:
We can make a Docker container (or docker-compose system) that runs both the PASS react app and a solid server out of the box, and maintains internally all necessary configuration and data. Then we will be able to easily deploy the full app to any environment.

---
### Rationale:
Ultimately, we will want to make it as easy as possible to set up PASS. If we have a well-written docker file, we can get a full PASS system (pod server, app, and env variables) up and running in a single command.

---
### Proposed Implementation (if applicable):

We can have a docker-compose file that builds 2 docker containers:

1. A pod server that runs community solid server
2. An NGINX server that serves the built React App

Using proper port forwarding, we can set up the docker-compose file to automatically set the env variables in the NGINX server to point the react app at the Pod server. We can also have the docker-compose automatically run letsencrypt to get certs to host the servers over https. This may require a third container (I haven't looked into that too closely yet).

The end result should be a system that keeps all information stored in the docker containers, with no leakage into the host environment (or if it needs to write to the host filesystem, writes to an area reserved to that container).

---
### Additional context:
We currently have a docker-compose file up on opencommons.net, but it is tied closely to the specifics of that droplet. It would be ideal if we can have a single, universal docker file that can be run on all systems.

### PR DESCRIPTION
**Pull request by Fokker50**
## CONCISE description of PR (PR Title)

---
## This PR:
Resolves #375 
 
1. Docker compose corrected:  external network was removed

---
## The files this PR effects:
docker-compose.yaml

### Components
List file names here

### Tests
Test can be be performed on remote server . Ensure Docker and Docker compose are installed.
You would need domain name pointing to your server IP address. Ensure that ports 80, 443 and 81, 3000 are open.(Configure Firewall to allow traffic on this ports)
1. Clone Repository to your machine , checkout to branch 375-docker-config.
2. Adjust config files with <your domain name>.
3.  Deploy
<docker-compose up -d>
4. Check
<docker-compose logs>
<docker ps> get the id of container and run <docker exec -it *cont_id*  sh>
5.  Access the Nginx Proxy manager UI via your IP: port 81 (example 10.0.0.0:81)
Sign in.
Email:    admin@example.com
Password: changeme 
6. Set up Proxy hosts: for react-app and solid-server (domain name, schema, forward-hostname or ip, port)
7. Access the services using domain names or IP and Port.
React-app IP:8080
Solid-server IP:3000

---
## Additional Context (optional):
Any feedback appreciated. 

---
## Future Steps/PRs Needed to Finish This Work (optional):
Need to fire out how solid-server shall be configured.               

---

### CODE REVIEW

**Comment by bingeboy** [comment]
Is this ready for review?

**Comment by Fokker50** [comment]
PR is ready for review

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      - "8080" # Expose port for other services
+    ports:
+      - "8080:80"
+    networks:
+      - nginx_default
+
+  solid-server:
+    image: solidproject/community-server:latest # Use updated image or create using Dockerfile
+    environment:
+      - BASE_URL=https://css.opencommons.net
```
Comment (timbot1789) [inline docker-compose.yaml]: The [environment variables section of the CSS readme](https://github.com/CommunitySolidServer/CommunitySolidServer#%EF%B8%8F-environment-variables) says that the environment variables used by Community solid server need to be prefixed with `CSS_`, so this should be `CSS_BASE_URL`.

**Review by timbot1789 — APPROVED** [review]
Tested this config on my own server. It's working well. We can customize it some more if we want, but this is a perfectly good core.

**Pull request merged**